packages/dashboard/src/store/actions/updateTableWidget/index.ts

```diff
@@ -0,0 +1,68 @@
+import { AssetQuery } from '@iot-app-kit/core';
+import { Action } from 'redux';
+import { DashboardState } from '~/store/state';
+import { ColumnDefinition, Item } from '@iot-app-kit/table';
+import { Widget } from '~/types';
+import { toId } from '@iot-app-kit/source-iotsitewise';
+
+type UpdateTableAssetPayload = {
+  widget: Widget;
+  assetQuery: AssetQuery[];
+};
+
+export interface UpdateTableAssetsAction extends Action {
+  type: 'UPDATE_TABLE_ASSET';
+  payload: UpdateTableAssetPayload;
+}
+
+export const onUpdateTableAssets = (payload: UpdateTableAssetPayload): UpdateTableAssetsAction => ({
+  type: 'UPDATE_TABLE_ASSET',
+  payload,
+});
+
+export const updateTableAssets = (state: DashboardState, action: UpdateTableAssetsAction): DashboardState => {
```

Comment:
I know this is the pattern we have been doing for other types of widgets with asset queries, but is there any way we can move this logic into the table component? I'm in the process of removing any store actions that don't apply to every widget. I don't want to add onto that work unless we absolutely have to.